cpp/models/abm/person.cpp

```diff
@@ -264,5 +263,20 @@ ScalarType Person::get_protection_factor(TimePoint t, VirusVariant virus, const
         t.days() - latest_protection.second.days());
 }
 
+void Person::add_test_result(TimePoint t, TestType type, bool result)
+{
+    TestResult test_result;
+    test_result.time_of_testing     = t;
+    test_result.type                = type;
+    test_result.is_allowed_to_enter = result;
+    // Remove outdated test results or replace the old result of the same type
+    m_test_results[{type}] = test_result;
```

Comment:
This can probably be done in a single line like `m_test_results[{type}] = {t, type, result};` or so.

Reply:
Yes, it is now optimised.